Common/CUDA/tvdenoising.cu

```diff
@@ -399,13 +404,14 @@ do { \
                 // d_src is the original image, with no change.
                 if (splits>1 & i>0){
 
-                    for (dev = 0; dev < deviceCount; dev++){   
+                    for (dev = 0; dev < deviceCount; dev++){
+                       
+                        is_last_chunk=!((sp*deviceCount+dev)<deviceCount*splits-1);
+                        is_first_chunk=!(sp*deviceCount+dev);
                         cudaSetDevice(gpuids[dev]);
                         cudaStreamSynchronize(stream[dev*nStream_device+1]);
-                        // Sometimes the last_chunk is smaller than the other ones, and thus if we don't reset the memory to zero, we'll gave ghosts residual data in the variable
-                        cudaMemsetAsync(d_u[dev], 0, mem_img_each_GPU,stream[dev*nStream_device+1]); 
-                        cudaMemcpyAsync(d_u [dev] +offset_device[dev], h_u +offset_host[dev],  bytes_device[dev]*sizeof(float), cudaMemcpyHostToDevice,stream[dev*nStream_device+1]);
-                       
+                        if (is_last_chunk) {cudaMemsetAsync(d_u[dev], 0, mem_img_each_GPU,stream[dev*nStream_device+1]);}
+                        cpy_from_host(d_u[dev],h_u,bytes_device[dev], offset_device[dev],offset_host[dev], pixels_per_slice, buffer_length, stream[dev*nStream_device+1],  is_first_chunk,  is_last_chunk, image_size);
                     }
 
                     for (dev = 0; dev < deviceCount; dev++){   

```

Comment:
Not using the cpy_from_host function to updata d_px, d_py,d_pz and d_src?

Reply:
Yes, its just here few lines below (e.g. line 430)